src/modules/navigator/geofence.cpp

```diff
@@ -118,6 +125,29 @@ bool Geofence::inside(const struct vehicle_global_position_s &global_position,
 
 bool Geofence::inside(double lat, double lon, float altitude)
 {
+	int32_t max_horizontal_distance = _param_max_hor_distance.get();
+	int32_t max_vertical_distance = _param_max_ver_distance.get();
+
+	if (max_horizontal_distance > 0 || max_vertical_distance > 0) {
+		bool updated = false;
+		orb_check(_home_position_sub, &updated);
+
+		if (updated) {
+			struct home_position_s home;
+			orb_copy(ORB_ID(home_position), _home_position_sub, &home);
+
+			float dist_xy = -1.0f;
```

Comment:
why is this inside the `if(updated)` ? the range should be checked on every call to this function I guess and not only if home updates
```